.github/workflows/build-docs.yml

```diff
@@ -0,0 +1,224 @@
+name: Build and deploy docs
+
+on:
+  # Trigger the workflow on push
+  push:
```

Comment:
Build docs on each push? We mostly do code changes and silly ruff fixes. Maybe have this run on master and develop only? If we need to write new docs and check them before pushing to develop, we can always trigger the docs rebuild manually.

Reply:
No, this is something I forgot to change after debugging the workflow. Thanks for noticing!